e2e/pull/pull.go

```diff
@@ -464,13 +465,107 @@ func (c ctx) testPullDisableCacheCmd(t *testing.T) {
 	}
 }
 
+// testPullUmask will run some pull tests with different umasks, and
+// ensure the output file hase the correct permissions.
+func (c ctx) testPullUmask(t *testing.T) {
+	umaskTests := []struct {
+		name       string
+		imagePath  string
+		umask      int
+		expectPerm uint32
+		force      bool
+	}{
+		{
+			name:       "0022 umask pull",
+			imagePath:  filepath.Join(c.env.TestDir, "0022-umask-pull"),
+			umask:      0022,
+			expectPerm: 0755,
+		},
+		{
+			name:       "0077 umask pull",
+			imagePath:  filepath.Join(c.env.TestDir, "0077-umask-pull"),
+			umask:      0077,
+			expectPerm: 0700,
+		},
+		{
+			name:       "0027 umask pull",
```

Comment:
Is a 3rd test telling us anything here? If we get perms right from 2 different umasks then our code is correct and we're really just testing upstream go functions I think?

Reply:
I dont think so, `0027` seems to be the magic umask where things start to break/not function. Example: https://github.com/sylabs/singularity/issues/2381